pkg/server/routes.go

```diff
@@ -1,16 +1,32 @@
 package server
 
 import (
+	"context"
 	"net/http"
 
 	"github.com/gorilla/mux"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sirupsen/logrus"
 
 	"github.com/netobserv/network-observability-console-plugin/pkg/handler"
+	"github.com/netobserv/network-observability-console-plugin/pkg/handler/auth"
 )
 
 func setupRoutes(cfg *Config) *mux.Router {
 	r := mux.NewRouter()
+	r.Use(func(orig http.Handler) http.Handler {
```

Comment:
If I understand correctly, all endpoints will be behind authentication.

We should exclude from authentication the /metrics endpoint and the fronted static files.

Reply:
hmm I think we still need to do that? We still don't want metrics or whatever being accessible from unauthenticated users?

Reply:
@OlivierCazade I've removed the metrics endpoint for the time being